src/contract_client/sent_transaction.ts

```diff
@@ -72,6 +72,17 @@ export class SentTransaction<T> {
 
   private send = async (): Promise<this> => {
     const timeoutInSeconds = this.assembled.options.timeoutInSeconds ?? DEFAULT_TIMEOUT
+    const timeoutTimestamp =
+      Math.floor(Date.now() / 1000) + timeoutInSeconds + 1;
+    this.assembled.built = TransactionBuilder.cloneFrom(this.assembled.built!, {
+      fee: this.assembled.built!.fee,
+      timebounds: {
+        minTime: 0,
+        maxTime: timeoutTimestamp,
+      },
+    })
+      .setSorobanData(new SorobanDataBuilder(this.assembled.simulationData.transactionData.toXDR("base64")).build())
```

Comment:
I think it should accept the non-base64 form: https://github.com/stellar/js-stellar-base/blob/master/src/sorobandata_builder.js#L56